### DISCUSSION TITLE
Group 1 Discussion Thread

### DISCUSSION THREAD

**Discussion by quarterfalls** [Digital-Preservation-Coalition/PythonStudyGroups#3] [General]
A space for Group 1 to discuss, troubleshoot and celebrate!

**Comment by quarterfalls** [comment]
Our group has agreed ahead of the next session to:

- Read through the primer ([wiki](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/wiki))
- Read through the [batch renaming how to](https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/Use%20Case%20-%20How%20to%20batch%20rename%20files.md)
- Think about a project from our own workplace that could be improved with Python (ideally a spreadsheet issue so we can use [PANDAS](https://pandas.pydata.org/))

And in the next session Nicola will give us an overview of how she has used ChatGPT to assist in writing a Python script

**Reply by quarterfalls** [reply]
And Melia has agreed to give us an overview of how she uses GitHub's Copilot to assist with writing code

**Reply by NicolaCaldwell** [reply]
I've used SublimeText and VSCode to write/run scripts

**Reply by quarterfalls** [reply]
I've also used SublimeText a little and have recently installed VSCode and Anaconda (but don't have much experience using them yet)

**Reply by MeliaJB** [reply]
My favorite tools for putting together and running Python scripts are Visual Studio Code (aka VS Code) with plugins such as:

- GitHub Copilot ( I believe there is a $10 subscription but they do offer a trial period in a pinch)
- GitHub Copilot Chat (allows you to have Chat GPT like conversations to figure out how to script)
- Python (debugger, tells you if there are any mistakes as you're scripting)
- Pylance (a python server for VS code)
- Jupyter (allows you to run Jupyter Notebook from VS Code)
If you don't use VS code, then Miniconda or Anaconda are good packages to install into your computer's shell environment that allow you to run Jupyter Notebook (and Python, I believe?)

**Reply by code-for-archives** [reply]
@quarterfalls Thanks for setting up this discussion thread.

**Comment by quarterfalls** [comment]
Hi all, I've just tried to access the bath renaming how to and am getting a 404 message so possibly it isn't available right now. Will try again next week :)

**Reply by lars-unimelb** [reply]
Hi, looks like the url was wrong (or it has been moved), the right one is: https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20-%20How%20to%20batch%20rename%20files.md

**Comment by MeliaJB** [comment]
Hi everyone, Sorry, but I won't be able to make tonight's session, due to a scheduling conflict that I completely forgot about. I can give the Git Hub Copilot walkthrough next month, or if need be, I can record a video later this week. I did find a brief overview on youtube for it as well: https://youtu.be/jXp5D5ZnxGM?si=OUbIxv5JItsMxU9i . See you next time!

**Comment by code-for-archives** [comment]
OFFICIAL

Thanks for letting us know, Melia.

I have gone to the Python meeting on Zoom but I have not been let in by the host yet. Has anyone else gotten in to the meeting?

Louise
OFFICIAL
From: Melia Jean-Baptiste ***@***.***>
Sent: Wednesday, February 28, 2024 9:41 AM
To: Digital-Preservation-Coalition/PythonStudyGroups ***@***.***>
Cc: Louise Preston ***@***.***>; Manual ***@***.***>
Subject: [EXT]Re: [Digital-Preservation-Coalition/PythonStudyGroups] Group 1 Discussion Thread (Discussion #3)

Caution: This email originated from outside of the organization! Do not click links, open attachments or reply, unless you recognize the sender's email address and know the content is safe!

Hi everyone, Sorry, but I won't be able to make tonight's session, due to a scheduling conflict that I completely forgot about. I can give the Git Hub Copilot walkthrough next month, or if need be, I can record a video later this week. I did find a brief overview on youtube for it as well: https://youtu.be/jXp5D5ZnxGM?si=OUbIxv5JItsMxU9i . See you next time!

-

**Reply by quarterfalls** [reply]
Hi Louise, Nicola and I have been let into the meeting and are in break out group 1
I'll ask one of the mentors to see if you can be let in
[https://us02web.zoom.us/j/81597329077?pwd=WnJEMm9veHNraGxBU2NVZHBMSHRzUT09](https://protect-au.mimecast.com/s/jQeFC1WZKqhpR3PkkFGK5kf?domain=us02web.zoom.us)

**Comment by quarterfalls** [comment]
Hey everyone, Nicola, Louise and I met today and Nicole gave us a great demo of her experience using ChatGPT. We agreed that next time we'll look at the file renamer example from the how tos: https://github.com/Digital-Preservation-Coalition/PythonStudyGroups/blob/main/How-Tos/Use%20Case%20-%20How%20to%20batch%20rename%20files.md

We also watched the GitHub copilot video Melia posted above (thanks for that). If you're up for talking us through how you use Copilot in the next session @MeliaJB I think that would still be useful :)

**Comment by MeliaJB** [comment]
Hi all, I was wondering if the meeting started for anyone else yet?

**Comment by code-for-archives** [comment]
UNOFFICIAL

Hi all,

I won't make it to the meeting today.

Melia, I have the meeting in my calendar as starting an hour later (10am AEDT).

Kind regards,

Louise
UNOFFICIAL
From: Melia Jean-Baptiste ***@***.***>
Sent: Wednesday, March 27, 2024 9:06 AM
To: Digital-Preservation-Coalition/PythonStudyGroups ***@***.***>
Cc: Louise Preston ***@***.***>; Manual ***@***.***>
Subject: [EXT]Re: [Digital-Preservation-Coalition/PythonStudyGroups] Group 1 Discussion Thread (Discussion #3)

EXTERNAL EMAIL: Do not click links or open attachments unless you trust the sender and know the content is safe.

Hi all, I was wondering if the meeting started for anyone else yet?

-

**Reply by MeliaJB** [reply]
Oh! Okay thanks for letting me know and see you next time then!

**Comment by MeliaJB** [comment]
Link to additional Python exercises: https://github.com/UMiamiLibraries/python4lib-python-exercises

**Reply by MeliaJB** [reply]
Also, here is the recording of today's GitHub Copilot [walkthrough](https://drive.google.com/file/d/1J-8KxefiwjPxLcRDjViAdBsv1KVZ2642/view?usp=share_link)

**Reply by quarterfalls** [reply]
Thanks so much for this demo Melia!!

**Comment by code-for-archives** [comment]
OFFICIAL

Sorry I couldn't make it, Melia and others.

Thanks for the recording!
OFFICIAL
From: Elise ***@***.***>
Sent: Wednesday, March 27, 2024 3:20 PM
To: Digital-Preservation-Coalition/PythonStudyGroups ***@***.***>
Cc: Louise Preston ***@***.***>; Manual ***@***.***>
Subject: [EXT]Re: [Digital-Preservation-Coalition/PythonStudyGroups] Group 1 Discussion Thread (Discussion #3)

EXTERNAL EMAIL: Do not click links or open attachments unless you trust the sender and know the content is safe.

Thanks so much for this demo Melia!!

-

**Comment by quarterfalls** [comment]
Hi everyone, sorry I missed today's session. We've just had a time change here in Australia and I messed up the times :|

**Reply by NicolaCaldwell** [reply]
No worries Elise - I would have done the same, the only reason I joined at the right time was because of Ellie's reminder email (we also had daylight savings change recently)! Louise and I caught up on the video that Melia made last time, so that was great :) I unfortunately won't be able to join the next couple of sessions as I'm away on leave but hopefully we can keep in contact after the course or join the next cohort too.

**Reply by quarterfalls** [reply]
Oh that's great! Yeah it would be great to stay in touch about what we're all working on. I tested Github Copilot myself in recent weeks and found it really easy to use directly in VS Code (although I probably won't pay for it once my free trial is up).

**Comment by code-for-archives** [comment]
OFFICIAL

Hi Elise,

Same issue here! I got a reminder from Ellie and that was how I knew it was changed and I had the time available in my calendar, so went there a bit late.

Nicola and I attended and she presented some work she has been doing. We also watched Melia's video from last time, which I think you saw first-hand!

Louise
OFFICIAL
From: Elise ***@***.***>
Sent: Wednesday, April 24, 2024 10:33 AM
To: Digital-Preservation-Coalition/PythonStudyGroups ***@***.***>
Cc: Louise Preston ***@***.***>; Manual ***@***.***>
Subject: [EXT]Re: [Digital-Preservation-Coalition/PythonStudyGroups] Group 1 Discussion Thread (Discussion #3)

EXTERNAL EMAIL: Do not click links or open attachments unless you trust the sender and know the content is safe.

Hi everyone, sorry I missed today's session. We've just had a time change here in Australia and I messed up the times :|

-

**Reply by quarterfalls** [reply]
Oh great, that's worked out!

**Comment by code-for-archives** [comment]
OFFICIAL

Hi all,

Sorry I missed the start time of 9am.

Aside from missing the time change, I have had some serious personal/family stuff going on lately and would not be able to contribute to this meeting. I would be interested to get an update on it.

I know there is only one session left this round. I hope to have more to provide for it by then.

Kind regards,

Louise
OFFICIAL
From: Elise ***@***.***>
Sent: Wednesday, April 24, 2024 10:42 AM
To: Digital-Preservation-Coalition/PythonStudyGroups ***@***.***>
Cc: Louise Preston ***@***.***>; Manual ***@***.***>
Subject: [EXT]Re: [Digital-Preservation-Coalition/PythonStudyGroups] Group 1 Discussion Thread (Discussion #3)

EXTERNAL EMAIL: Do not click links or open attachments unless you trust the sender and know the content is safe.

Oh that's great! Yeah it would be great to stay in touch about what we're all working on. I tested Github Copilot myself in recent weeks and found it really easy to use directly in VS Code (although I probably won't pay for it once my free trial is up).

-
